thesis-de/content-de.tex

```diff
@@ -2539,7 +2539,223 @@ \section{Prototyp 2}
+diesem Projekt für die Metric-Daten jedes virtuellen Servers genutzt wird,
+wurde zusammen mit allen Projektmitgliedern definiert und abgesprochen.
+Dabei wurde sich auf eine spezielle Abspeicherung der Daten geeinigt. Es
+handelt sich um die Verwendung von einer einzelnen Datentabelle, welche
+nur zwei Attribute besitzt. Bei beiden Attributen handelt es sich um ein
+JSON Objekt. Ersteres beinhaltet Metadaten zu dem einzelnen Datensatz.
+Das ist zum Beispeil eine generierte ID, um den Datensatz spezifisch
+selektieren zu können, oder die Informationen seit wann der Datensatz
+in der Datenhaltung existiert, bzw. wann und von welchem System angefasst
+wurde. Zweites JSON Objekt/Attribut beinhaltet den eigentlichen Datensatz,
+also die werthaltigen Informationen, welche auch später analysiert werden.
+Die Inhalte des Attributes werden klar in dem Metadatensatz definiert.
+Der Vorteil dieser Datenspeicherung ist zum einen die Effizienz des
+Speicherplatzes und Geschwindigkeit und zum anderen das Zusammenhalten der
+zugehörigen Informationen. Desweiteren können in diesem Model sehr einfach neue
+Informationen hinzugefügt werden, indem diese in dem Metadatensatz definiert
+werden. Dies ist ein Vorteil der Skalierbarkeit.
+
+Da sich bei diesem Model die Geschwindigkeit der Datenabfragen mit der Menge
+der Daten in einer Tabelle verlangsamt, wird die automatische Partitionierung
+angewandt, welche im nachfolgendem Punkt~\ref{subsec:postgres_partition}
+beschrieben wird.
+\nl%
+
+\subsection{Partitionierung}
+\label{subsec:postgres_partition}
+Bei der Partitionierung in dem Postgres-System muss zwischen zwei Arten von
+Partitionierung entschieden werden. Ersteres ist die manuelle Partitionierung
+durch den Administrator. Dabei werden über einen Index Datenbereiche definiert,
+welche logisch voneinander getrennt werden. Mit nachfolgendem Code-Prototyp
+kann solch ein Trennung im groben vom Administration definiert werden.
+
+\begin{listing}
+  \inputminted{sql}{../listings/postgres-part-sample.txt}
+  \caption{Manuelle Partitionierung Postgres}
+\end{listing}
+
+In diesem Prototypen wird eine Haupttabelle in zwei Kinder-Tabellen aufgeteilt.
+Diese Kinder-Tabellen halten dabei bestimmte definierte Datenbereiche. In
+diesem Prototypen enthält die Kinder-Tabelle \texttt{measurement_xMINUS12Hour}
+nur Datensätze bei welchen der Zeitstempel der beinhalteten Daten zwischen dem
+aktuellen Zeitpunkt und dem aktuellen Zeitpunkt minus 12 Stunden liegt. Die
+zweite Kinder-Tabelle beinhaltet wiederrum nicht die Daten von der ersten
+Kinder-Tabelle, sondern nur alle Daten welche einen Zeitstempel innerhalb
+eines Tages besitzen. Diese Art von Partitionierung kann auf jedem beliebigen
+Attribut angewendet werden, sodass zum Beispiel auch noch CPU-Daten von
+RAM-Daten getrennt werden. Für die vollständige Implementierung einer manuellen
+Partitionierung werden jedoch noch weitere Funktionen benötigt, auf welche
+jedoch nicht im Detail eingegangen werden.
```

Comment:
das ist etwas unglücklich. kannst du den satz streichen? entweder wir ignorieren das komplett oder erwähnen in ein-zwei sätzen was noch fehlt udn stempeln das als lapidar ab.